test: TestStreamWriterAttachOutgoingConn

```
--- FAIL: TestStreamWriterAttachOutgoingConn (0.01s)
	stream_test.go:65: #0: working status = false, want true
2016-08-20 00:04:55.732243 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.732321 I | rafthttp: peer 1 became active
2016-08-20 00:04:55.732368 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-08-20 00:04:55.732453 E | rafthttp: failed to write 1 on stream Message (blah)
2016-08-20 00:04:55.732487 I | rafthttp: peer 1 became inactive
2016-08-20 00:04:55.734615 W | rafthttp: lost the TCP streaming connection with peer 1 (stream Message writer)
2016-08-20 00:04:55.746223 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.768347 I | rafthttp: started streaming with peer 2 (stream Message reader)
2016-08-20 00:04:55.778795 I | rafthttp: peer 2 became active
2016-08-20 00:04:55.778891 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
2016-08-20 00:04:55.778965 W | rafthttp: lost the TCP streaming connection with peer 2 (stream Message reader)
2016-08-20 00:04:55.779053 I | rafthttp: stopped streaming with peer 2 (stream Message reader)
2016-08-20 00:04:55.795582 I | rafthttp: started streaming with peer 2 (stream Message reader)
2016-08-20 00:04:55.795799 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.802469 I | rafthttp: peer 1 became active
2016-08-20 00:04:55.802531 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-08-20 00:04:55.802681 I | rafthttp: peer 2 became active
2016-08-20 00:04:55.802726 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
2016-08-20 00:04:55.803567 W | rafthttp: lost the TCP streaming connection with peer 2 (stream Message reader)
2016-08-20 00:04:55.803626 E | rafthttp: failed to read 2 on stream Message (net/http: request canceled)
2016-08-20 00:04:55.803659 I | rafthttp: peer 2 became inactive
2016-08-20 00:04:55.803710 I | rafthttp: stopped streaming with peer 2 (stream Message reader)
2016-08-20 00:04:55.804845 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.804978 I | rafthttp: started streaming with peer 2 (stream Message reader)
2016-08-20 00:04:55.805943 I | rafthttp: peer 1 became active
2016-08-20 00:04:55.805993 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-08-20 00:04:55.806129 I | rafthttp: peer 2 became active
2016-08-20 00:04:55.806172 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
2016-08-20 00:04:55.806602 W | rafthttp: lost the TCP streaming connection with peer 2 (stream Message reader)
2016-08-20 00:04:55.806659 E | rafthttp: failed to read 2 on stream Message (net/http: request canceled)
2016-08-20 00:04:55.806692 I | rafthttp: peer 2 became inactive
2016-08-20 00:04:55.806739 I | rafthttp: stopped streaming with peer 2 (stream Message reader)
2016-08-20 00:04:55.807932 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.808051 I | rafthttp: started streaming with peer 2 (stream MsgApp v2 reader)
2016-08-20 00:04:55.817888 I | rafthttp: peer 2 became active
2016-08-20 00:04:55.817974 I | rafthttp: established a TCP streaming connection with peer 2 (stream MsgApp v2 reader)
2016-08-20 00:04:55.819084 I | rafthttp: peer 1 became active
2016-08-20 00:04:55.819166 I | rafthttp: established a TCP streaming connection with peer 1 (stream MsgApp v2 writer)
2016-08-20 00:04:55.819692 W | rafthttp: lost the TCP streaming connection with peer 2 (stream MsgApp v2 reader)
2016-08-20 00:04:55.819766 I | rafthttp: stopped streaming with peer 2 (stream MsgApp v2 reader)
2016-08-20 00:04:55.820799 I | rafthttp: closed the TCP streaming connection with peer 1 (stream MsgApp v2 writer)
2016-08-20 00:04:55.820854 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.822106 I | rafthttp: closed the TCP streaming connection with peer 1 (stream Message writer)
2016-08-20 00:04:55.822164 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.823609 I | rafthttp: closed the TCP streaming connection with peer 1 (stream Message writer)
2016-08-20 00:04:55.823668 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.824954 I | rafthttp: starting peer 1...
2016-08-20 00:04:55.825409 I | rafthttp: started HTTP pipelining with peer 1
2016-08-20 00:04:55.840841 I | rafthttp: started peer 1
2016-08-20 00:04:55.841055 I | rafthttp: added peer 1
2016-08-20 00:04:55.841154 I | rafthttp: stopping peer 1...
2016-08-20 00:04:55.843536 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.843636 I | rafthttp: started streaming with peer 1 (stream Message reader)
2016-08-20 00:04:55.845441 I | rafthttp: started streaming with peer 1 (stream MsgApp v2 reader)
2016-08-20 00:04:55.845831 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.845904 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.845987 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.846119 I | rafthttp: stopped HTTP pipelining with peer 1
2016-08-20 00:04:55.846197 I | rafthttp: stopped streaming with peer 1 (stream MsgApp v2 reader)
2016-08-20 00:04:55.846301 I | rafthttp: stopped streaming with peer 1 (stream Message reader)
2016-08-20 00:04:55.846362 I | rafthttp: stopped peer 1
2016-08-20 00:04:55.846684 I | rafthttp: starting peer 1...
2016-08-20 00:04:55.847089 I | rafthttp: started HTTP pipelining with peer 1
2016-08-20 00:04:55.853136 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.854852 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.871688 I | rafthttp: started streaming with peer 1 (stream MsgApp v2 reader)
2016-08-20 00:04:55.872030 I | rafthttp: started streaming with peer 1 (stream Message reader)
2016-08-20 00:04:55.875786 I | rafthttp: started peer 1
2016-08-20 00:04:55.875978 I | rafthttp: added peer 1
2016-08-20 00:04:55.876037 I | rafthttp: stopping peer 1...
2016-08-20 00:04:55.876178 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.876269 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.876426 I | rafthttp: stopped HTTP pipelining with peer 1
2016-08-20 00:04:55.876506 I | rafthttp: stopped streaming with peer 1 (stream MsgApp v2 reader)
2016-08-20 00:04:55.876588 I | rafthttp: stopped streaming with peer 1 (stream Message reader)
2016-08-20 00:04:55.876646 I | rafthttp: stopped peer 1
2016-08-20 00:04:55.876702 I | rafthttp: removed peer 1
2016-08-20 00:04:55.877177 I | rafthttp: updated peer 1
2016-08-20 00:04:55.877446 I | rafthttp: starting peer 1...
2016-08-20 00:04:55.877751 I | rafthttp: started HTTP pipelining with peer 1
2016-08-20 00:04:55.884562 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.885983 I | rafthttp: started streaming with peer 1 (writer)
2016-08-20 00:04:55.891577 I | rafthttp: started peer 1
2016-08-20 00:04:55.891910 I | rafthttp: added peer 1
2016-08-20 00:04:55.895580 I | rafthttp: started streaming with peer 1 (stream MsgApp v2 reader)
2016-08-20 00:04:55.896244 I | rafthttp: started streaming with peer 1 (stream Message reader)
2016-08-20 00:04:55.902519 I | rafthttp: stopping peer 1...
2016-08-20 00:04:55.902656 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.902742 I | rafthttp: stopped streaming with peer 1 (writer)
2016-08-20 00:04:55.902891 I | rafthttp: stopped HTTP pipelining with peer 1
2016-08-20 00:04:55.902995 I | rafthttp: stopped streaming with peer 1 (stream MsgApp v2 reader)
2016-08-20 00:04:55.903102 I | rafthttp: stopped streaming with peer 1 (stream Message reader)
2016-08-20 00:04:55.903161 I | rafthttp: stopped peer 1
FAIL
```

--- a/rafthttp/stream_test.go
+++ b/rafthttp/stream_test.go
@@ -59,11 +59,10 @@ func TestStreamWriterAttachOutgoingConn(t *testing.T) {
 			}
 		}
 
-		// msgc has been swapped with a new one now that prevwfc is closed
-		msgc, ok := sw.writec()
-		if !ok {
-			t.Errorf("#%d: working status = %v, want true", i, ok)
-		}
+		// if prevwfc != nil, the new msgc is ready since prevwfc has closed
+		// if prevwfc == nil, the first connection may be pending, but the first
+		// msgc is already available since it's set on calling startStreamwriter
+		msgc, _ := sw.writec()
 		msgc <- raftpb.Message{}
 
 		select {

